src/metar/metar_csv2ioda.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env python
+
+#
+# (C) Copyright 2020 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# author: Greg Thompson gthompsn AT ucar DOT edu
+# This script will work with decoded, CSV files created by gthompsn stored at NCAR
+#   data-access.ucar.edu:/glade/campaign/ral/aap/gthompsn/METARs/2019/20191231/2019123118_metars.csv.gz
+#
+#..Temporarily include the ioda_conv_ncio library path explicitly using this: export PYTHONPATH=../lib-python:$PYTHONPATH
+#
+
+from __future__ import print_function
+import sys
+import math
+from datetime import datetime
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+from pathlib import Path
+from csv import DictReader
+import netCDF4
+
+IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
+if not IODA_CONV_PATH.is_dir():
+    IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
+sys.path.append(str(IODA_CONV_PATH.resolve()))
+
+import meteo_utils
+import ioda_conv_ncio as iconv
+from orddicts import DefaultOrderedDict
+
+class reformatMetar(object):
+
+    def __init__(self, filename, date):
+
+        self.filename = filename
+        self.date = date
+        self.meteo_utils = meteo_utils.meteo_utils()
+
+        # Read in CSV-formatted file of METAR data
+        self._rd_metars()
+
+        return
+
+    def _rd_metars(self):
+
+        # data is the dictionary of incoming observation (METAR) data
+        data = {}
+
+        data['ob_icao'] = []
+        data['ob_lat'] = []
+        data['ob_lon'] = []
+        data['ob_time'] = []
+        data['ob_datetime'] = []
+        data['ob_elev'] = []
+        data['ob_psfc'] = []
+        data['ob_temp'] = []
+        data['ob_spfh'] = []
+        data['ob_uwnd'] = []
+        data['ob_vwnd'] = []
+
+        '''
+        Read in the METARs data
+        Header contains: ICAO, Unix_time, Latitude, Longitude, Elev, MSLP, Altimeter, Temp, Dewp, Wdir, Wspd
+        '''
+
+        # open file in read mode
+        with open(self.filename, 'r') as fh:
+            # pass the file object to reader() to get the reader object
+            csv_dict_reader = DictReader(fh)
+            column_names = csv_dict_reader.fieldnames
+            print(column_names)
+            # Iterate over each row in the csv using reader object
+            for row in csv_dict_reader:
+                # row variable is a list that represents a row in csv
+
+                if row['ICAO'] is '':
+                    continue
+                else:
+                    icao = str(row['ICAO'])
+                try:
+                    utime = int(row['Unix_time'])
+                except:
```

Comment:
if you can write the specific exceptions expected here, that is preferred I think.

Reply:
like TypeError, ValueError, etc.

Reply:
Also, I think you can put all of the lines that are try: except: continue into one try and except. Python will continue at the first failed try.

Reply:
As a newcomer to Python programming, I don't fully follow what you're recommending. Is the solution here simply to put the consecutive "try" pieces into a single block as opposed to each one being on a separate try/except chunk?

Reply:
Yes, if the except for each case is "continue", just do:
try:
  line1
  line2
  line3
except:
  continue

it will go to the continue the first time it encounters an error. You would only want separate try statements if the exceptions were going to be different, or if the action to perform in the except was different.

Reply:
OK, thanks.  I've changed my code (will push in branch once I resolve a batch of your suggestions).  And thanks for this help!!!